src/mlpack/core/cv/metrics/r2_score.hpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file r2_score.hpp
+ * @author Bisakh Mondal
+ *
+ * The R^2 (Coefficient of determination) regression metric.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_METRICS_R2SCORE_HPP
+#define MLPACK_CORE_CV_METRICS_R2SCORE_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace cv {
+
+/**
+ * The R2Score is a metric of performance for regression algorithms
+ * that represents the proportion of variance (of y) that has been
+ * explained by the independent variables in the model. It provides
+ * an indication of goodness of fit and therefore a measure of how
+ * well unseen samples are likely to be predicted by the model,
+ * through the proportion of explained variance.
+ * As R2Score is dataset dependent it can have wide range of values. The
+ * best possible score is @f$R^2 =1.0@f$. Values of R2 outside the range
+ * 0 to 1 can occur when the model fits the data worse than a horizontal
+ * hyperplane. This would occur when the wrong model was chosen, or
+ * nonsensical constraints were applied by mistake. A model which
+ * predicts exactly the expected value of y, disregarding the input
+ * features, gets a R2Score equals to 0.0.
```

Comment:
Rather than adding this, If you have a paper that advocates this model that would be better. Let me know what you think.

Reply:
It's a very common, widely used statistical metric, almost every newbie is acquainted with it. I don't think adding paper will be necessary. I appreciate your suggestions.Thanks